controllers/flinkcluster_submit_job_script.go

```diff
@@ -49,7 +49,7 @@ function list_jobs() {
 function check_existing_jobs() {
 	echo "Checking existing jobs..."
 	list_jobs
-	if list_jobs | grep -e "(SCHEDULED)" -e "(RUNNING)" -e "(FINISHED)" -e "(FAILED)"; then
+	if list_jobs | grep -e "(SCHEDULED)" -e "(RUNNING)"; then
```

Comment:
This reminded me that we might need to review the logic of extracting Flink job ID [here](https://github.com/GoogleCloudPlatform/flink-on-k8s-operator/blob/a8186417d92c3eab1088570baf1d23fd5e8bbfc6/controllers/flinkcluster_observer.go#L250).  We don't allow multiple jobs in RUNNING state at the same time, but when all of them are in failed/canceled/succeeded state, how do we determine which one is the current one? I think we might need to use the recorded flink job id as a reference.

Reply:
Yes, we need to improve this later if we support cluster updating by  update API.